tests/baselines/reference/api/typescript.d.ts

```diff
@@ -3765,10 +3765,10 @@ declare namespace ts {
         updateJSDocReadonlyTag(node: JSDocReadonlyTag, tagName: Identifier | undefined, comment: string | NodeArray<JSDocComment> | undefined): JSDocReadonlyTag;
         createJSDocUnknownTag(tagName: Identifier, comment?: string | NodeArray<JSDocComment>): JSDocUnknownTag;
         updateJSDocUnknownTag(node: JSDocUnknownTag, tagName: Identifier, comment: string | NodeArray<JSDocComment> | undefined): JSDocUnknownTag;
-        createJSDocDeprecatedTag(tagName: Identifier, comment?: string | NodeArray<JSDocComment>): JSDocDeprecatedTag;
-        updateJSDocDeprecatedTag(node: JSDocDeprecatedTag, tagName: Identifier, comment?: string | NodeArray<JSDocComment>): JSDocDeprecatedTag;
-        createJSDocOverrideTag(tagName: Identifier, comment?: string | NodeArray<JSDocComment>): JSDocOverrideTag;
-        updateJSDocOverrideTag(node: JSDocOverrideTag, tagName: Identifier, comment?: string | NodeArray<JSDocComment>): JSDocOverrideTag;
+        createJSDocDeprecatedTag(tagName: Identifier | undefined, comment?: string | NodeArray<JSDocComment>): JSDocDeprecatedTag;
```

Comment:
I don't get this one at all - can you end up with a `deprecated` tag which would print out with a different tag name?

```ts
/**
 * @dagnabit
 */
```

Reply:
I don't get it either, but all of the other tags are written like this. It looks like there's a helper `parseSimpleTag` that states that the argument could be undefined, but in practice it never actually calls any of these without a tag, so I'm betting they all can drop the `| undefined` and work, besides the API break.

Reply:
No, I'm wrong, they are undefined because they have defaults and loads of callers pass undefined. I'm not really sure what the deal is.

Reply:
I'm going to resolve this thread in favor of the other one about this just so I can get the list of TODOs down; the gist is that many of these JSDoc tags have alternatives, so take an identifier, and then undefined means "give me whatever is the default one people usually use". Whether or not that's good I'm not sure (but I want to merge this thread with the other one; this one is broken thanks to the module rebase).